pkg/ctl/utils/update_cluster_endpoint_access.go

```diff
@@ -0,0 +1,140 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/kris-nova/logger"
+	"github.com/spf13/pflag"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+)
+
+func updateClusterEndpointsCmd(cmd *cmdutils.Cmd) {
+	cfg := api.NewClusterConfig()
+	cmd.ClusterConfig = cfg
+
+	cmd.SetDescription("update-cluster-endpoints", "Update Kubernetes API endpoint access configuration", "")
+
+	cmd.FlagSetGroup.InFlagSet("General", func(fs *pflag.FlagSet) {
+		cmdutils.AddNameFlag(fs, cfg.Metadata)
+		cmdutils.AddRegionFlag(fs, cmd.ProviderConfig)
+		cmdutils.AddConfigFileFlag(fs, &cmd.ClusterConfigFile)
+		cmdutils.AddApproveFlag(fs, cmd)
+		cmdutils.AddTimeoutFlag(fs, &cmd.ProviderConfig.WaitTimeout)
+	})
+
+	var private, public bool
+	cmd.FlagSetGroup.InFlagSet("Update private/public Kubernetes API endpoint access configuration",
+		func(fs *pflag.FlagSet) {
+			fs.BoolVar(&private, "private-access", false, "access for private (VPC) clients")
+			fs.BoolVar(&public, "public-access", true, "access for public clients")
+		})
+	cmd.SetRunFuncWithNameArg(func() error {
```

Comment:
i think this should be `cmd.SetRunFunc(func() error {` since we are not actually looking at the `Cmd#NameArg` here